### PR TITLE
Fixes #9718: Allow devel installer to use --reset

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Anyone able to give this some review love?

File: hooks/boot/01-helpers.rb [hunk 1]
```diff
@@ -20,18 +20,30 @@ def execute(commands)
...
 
       !results.include? false
     end
+
+    def command_exists?(command)
+      `which #{command} > /dev/null 2>&1`
+      $?.success?
+    end
+
+    def service_exists?(service)
```
Comment (stbenjam) [inline hooks/boot/01-helpers.rb]: I think this is misnamed, it's really checking if a process is running.

File: hooks/pre/10-reset_feature.rb [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
-  $?.success?
+def service
+  Kafo::Helpers.command_exists?('service-wait') ? 'service-wait' : 'service'
 end
 
 def stop_services
   Kafo::KafoConfigure.logger.info 'Ensuring services httpd and foreman-tasks are stopped.'
-  Kafo::Helpers.execute('service httpd stop && service foreman-tasks stop')
+  Kafo::Helpers.execute("#{service} httpd stop")
+  Kafo::Helpers.execute("#{service} foreman-tasks stop") if Kafo::Helpers.service_exists?('foreman-tasks')
```
Comment (stbenjam) [inline hooks/pre/10-reset_feature.rb]: Can we just use the new katello-service tool? It'll stop whatever's available, `katello-service --exclude mongod,postgresql` I think would do what you want and all the service commands could be removed from the other sections.

File: hooks/boot/01-helpers.rb [hunk 3]
```diff
@@ -20,18 +20,30 @@ def execute(commands)
...
+      `ps cax | grep #{service} > /dev/null 2>&1`
```
Comment (ekohl) [inline hooks/boot/01-helpers.rb]: Do you really trust ps and grep for this?

File: hooks/pre/10-reset_feature.rb (hunk 2 above)
Comment (ehelms) [inline hooks/pre/10-reset_feature.rb]: I think I can go that route -- I'll have to do an update to the devel puppet module to ensure katello-service gets installed first.

**Comment by beav** [comment]
@ehelms is this still needed?

**Comment by ehelms** [comment]
Still relevant but largely outdated

**Pull request closed without merging**